src/compiler/gcc.rs

```diff
@@ -435,6 +437,11 @@ where
             },
         };
 
+        too_hard_for_direct_mode = match arg.flag_str() {
+            Some(s) => s == "-Xpreprocessor",
```

Comment:
shouldn't -Wp,* be excluded too?

Reply:
It should, but I was under the impression that it's not even supported by `sccache` at all? Maybe I misunderstood. Is it as simple as doing `arg.flag_str().starts_with("-Wp,")`?

Reply:
We should probably add -Wp in ARGS. Probably something like `take_arg!("-Wp", OsString, Concatenated(','), PreprocessorArgument)`

I guess you could put this check for -Xpreprocessor/-Wp where ProcessorArgument are handled, rather than for all args. Then there's the question about the second loop for Xclangs, and, how @-files are handled with direct mode.